### ISSUE TITLE
"Unification incomplete for case" error message could be improved

### ISSUE THREAD

**Issue by JonathanAldrich** [boyland/sasylf#116]
Due to a typo I put the wrong metavariable inside [brackets] in a case rule construct.  This resulted in a "Unification incomplete for case" error message.

Here's the offending code:
```
                case rule
                    d6: e3 => e5
                    d7: e4[e3'] => e1'
                    ----------------------------- eval-let
                    _: let x = e3 in e4[x] => e1'
                is
                    proof by unproved
                end case

```
based on the obvious big-step rule for let.  Obviously the e3' should be e5 here...but the typo was a bit hard to find.

It's probably genuinely hard to produce a good error message in this case, but I wonder if a bit of engineering could at least pinpoint e3' as the cause of the problem?  In this case e3' was already bound, and e5 is fresh, so it should be possible in principle to detect that e3' is the problem.

Not super high priority but this is the kind of thing we should fix to make SASyLF more usable.

**Comment by boyland** [comment]
True.  Incomplete unification is difficult technically to handle because the failure doesn't give any information.  But it should be possible to rummage around and guess possible problems.

**Comment by JonathanAldrich** [comment]
I put another example of this problem in the error-examples Google Drive folder, shared with John Boyland.  I'm not putting this on GitHub because it contains a partial solution to homework exercises.

**Comment by boyland** [comment]
[like]  John T Boyland reacted to your message:
________________________________

**Comment by Rakshitha-Ireddi** [comment]
I’ve opened a PR that addresses this:  <https://github.com/boyland/sasylf/pull/138>

**What it does:** When incomplete unification happens in the second case-analysis unification (the path that previously only printed the generic “Unification incomplete for case” line), the checker now attaches the same kind of **subexpression hint** as the first unification path, using `TermPrinter`. If the failing subterm is a metavariable **already fixed** by an earlier premise, the message notes that so typos like the wrong metavariable in brackets are easier to spot.

Happy to adjust wording or heuristics based on feedback.
